### PR TITLE
netty: Netty server poorly handles unknown content type

### PR DESCRIPTION
**Pull request by ramaraochavali**
PR for https://github.com/grpc/grpc-java/issues/3370

### CODE REVIEW

File: netty/src/main/java/io/grpc/netty/NettyServerHandler.java [hunk 1]
```diff
@@ -375,15 +378,40 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
...
+
+      if (!HTTP_METHOD.equals(headers.method())) {
+        respondWithHttpError(ctx, streamId, 405, Status.Code.INTERNAL,
+            String.format("Method '%s' is not supported", headers.method()));
+        return;
+      }
+      // Remove the leading slash of the path and get the fully qualified method name
+      CharSequence path = headers.path();
+      if (path.charAt(0) != '/') {
+        respondWithHttpError(ctx, streamId, 404, Status.Code.UNIMPLEMENTED,
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/NettyServerHandler.java]: The 404 case should be first. If both content-type and path are wrong, we should respond with 404.

**Comment by ramaraochavali** [comment]
moved the path check up to return 404 early. Added explicit `path==null` to give a better message otherwise we could have combined both.

File: netty/src/main/java/io/grpc/netty/NettyServerHandler.java [hunk 2]
```diff
@@ -375,15 +378,49 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
...
+            String.format("Expected path to start with /: %s", path));
+        return;
+      }
+
+      String method = path.subSequence(1, path.length()).toString();
+
       // Verify that the Content-Type is correct in the request.
-      verifyContentType(streamId, headers);
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/NettyServerHandler.java]: Oops. I didn't delete these methods. They're now unused: `determineMethod`, `verifyContentType`

File: netty/src/main/java/io/grpc/netty/NettyServerHandler.java [hunk 3]
```diff
@@ -375,15 +378,49 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
...
+        return;
+      }
+
+      if (path.charAt(0) != '/') {
+        respondWithHttpError(ctx, streamId, 404, Status.Code.UNIMPLEMENTED,
+            String.format("Expected path to start with /: %s", path));
+        return;
+      }
+
+      String method = path.subSequence(1, path.length()).toString();
```
[the remark this answers is not in the document]
Comment (ejona86) [reply netty/src/main/java/io/grpc/netty/NettyServerHandler.java]: `path` is a CharSequence, so it doesn't have `substring`.

File: netty/src/main/java/io/grpc/netty/NettyServerHandler.java (hunk 2 above)
Comment (ejona86) [reply netty/src/main/java/io/grpc/netty/NettyServerHandler.java]: I went ahead and deleted them, since it was trivial.

**Comment by ejona86** [comment]
@ramaraochavali, thank you!

File: netty/src/main/java/io/grpc/netty/NettyServerHandler.java (hunk 2 above)
Comment (ramaraochavali) [reply netty/src/main/java/io/grpc/netty/NettyServerHandler.java]: @ejona86  great..thanks..I should have done it my self. Did not pay attention. Sorry :-)

**Pull request merged**